packages/docregistry/test/context.spec.ts

```diff
@@ -85,10 +85,15 @@ describe('docregistry/context', () => {
     describe('#fileChanged', () => {
       it('should be emitted when the file is saved', async () => {
         const path = context.path;
+        const hash = context.contentsModel?.hash ?? null;
         let called = false;
         context.fileChanged.connect((sender, args) => {
           expect(sender).toBe(context);
           expect(args.path).toBe(path);
+          if (hash) {
+            expect(args.hash).not.toBeNull();
+            expect(args.hash).not.toBe(hash);
+          }
```

Comment:
I think that rather than using a conditional expect a new dedicated test would be desirable. In particular such a test should verify that the `context.fileChanged` is not emitted. It will need some work to mock the server response here I believe.